wallets/dcr/syncnotification.go

```diff
@@ -0,0 +1,678 @@
+package dcr
+
+import (
+	"math"
+	"time"
+
+	"github.com/planetdecred/dcrlibwallet/spv"
+	"golang.org/x/sync/errgroup"
+)
+
+func (w *Wallet) spvSyncNotificationCallbacks() *spv.Notifications {
+	return &spv.Notifications{
+		PeerConnected: func(peerCount int32, addr string) {
+			w.handlePeerCountUpdate(peerCount)
+		},
+		PeerDisconnected: func(peerCount int32, addr string) {
+			w.handlePeerCountUpdate(peerCount)
```

Comment:
This is really confusing how `PeerConnected` and `PeerDisconnected` execute the same function with no changes to inputs. It be easier to just execute one since both of them might not return any difference.

Reply:
reducing this unnecessary calls would reduce the number of mutex blocks requested especially on functions that require mutex blocks.

Reply:
Also if only you exported all those functions in `type Notification struct{}`, this whole implementation wouldn't be necessary that is unless there is something I am missing out on.